### PR TITLE
Fix for required properties validation for inline objects (ObjectProperty), small bugfix, test improvement

### PR DESCRIPTION
**Pull request by giuliopulina**
This is more a workaround due to the fact that swagger-core doesn't retain original required properties (https://github.com/swagger-api/swagger-core/issues/2479) and they are not going to change this behavior.

### CODE REVIEW

**Comment by giuliopulina** [comment]
need to check why builds are failing here but not on my fork :(

**Comment by giuliopulina** [comment]
fixed 👍

**Comment by giuliopulina** [comment]
@sylvainlaurent the PR is ready, may I ask to review it when you can? I would like to use the new version in my project

File: src/main/java/com/github/sylvainlaurent/maven/swaggervalidator/doc/traversal/node/ObjectPropertyRequiredStore.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+import static java.util.Collections.synchronizedMap;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ObjectPropertyRequiredStore {
+
+    private static final Map<String, List<String>> requiredObjectPropertiesMap = synchronizedMap(new HashMap<String, List<String>>());
```
Comment (sylvainlaurent) [inline src/main/java/com/github/sylvainlaurent/maven/swaggervalidator/doc/traversal/node/ObjectPropertyRequiredStore.java]: you'd better use a `java.util.IdentityHashMap` which is made for the use case you have here and is actually safer : the key you create for the Map could have some collisions if 2 objects have the identityHashCode.

Comment (giuliopulina) [reply src/main/java/com/github/sylvainlaurent/maven/swaggervalidator/doc/traversal/node/ObjectPropertyRequiredStore.java]: @sylvainlaurent thanks for the advice, I didn't know about java.util.IdentityHashMap, always good to learn something new ;)

**Comment by sylvainlaurent** [comment]
Merged. I guess you'd like a release with it. I'll do one this evening, CH time.

**Comment by giuliopulina** [comment]
@sylvainlaurent fine for me, thanks a lot!

**Comment by sylvainlaurent** [comment]
@giuliopulina : it's released in 1.1.1 which will appear in maven central in a few hours.

**Comment by giuliopulina** [comment]
@sylvainlaurent thanks again ;)

**Pull request merged**